pandas/core/frame.py

```diff
@@ -588,14 +588,16 @@ def __repr__(self):
             return buf.getvalue()
 
         max_rows = get_option("display.max_rows")
+        min_rows = get_option("display.min_rows")
         max_cols = get_option("display.max_columns")
         show_dimensions = get_option("display.show_dimensions")
         if get_option("display.expand_frame_repr"):
             width, _ = console.get_console_size()
         else:
             width = None
-        self.to_string(buf=buf, max_rows=max_rows, max_cols=max_cols,
-                       line_width=width, show_dimensions=show_dimensions)
+        self.to_string(buf=buf, max_rows=max_rows, min_rows=min_rows,
```

Comment:
can the logic be added before this call to `.to_string` (isn't it as simple as just set max_rows to min_rows if len(self.frame) > max_rows?)

Reply:
For the simple cases, yes. And I agree that could be a nice option. But there are some corner cases that are only handled within the `DataFrameFormatter` class (eg if max_rows = 0, we check the terminal size)

Reply:
Although, looking at it more closely, that special case is only if `max_rows` is 0 or None, so that could be easily checked. So yes, this would in principle be possible (and probably even give easier code). But that means that we move some of the logic of the repr out of the Formatter class.

Reply:
> But that means that we move some of the logic of the repr out of the Formatter class.

i agree. but this logic is related to display options. i'd prefer to see all the display option handling removed from the Formatter classes entirely and only appear in `__repr__` and `_repr_html_`

it appears from the comments that adding arguments to `to_string` is not considered undesirable. so this topic can be addressed another day.